bokehjs/src/lib/models/widgets/tables/data_cube.ts

```diff
@@ -222,11 +233,10 @@ export class DataCubeProvider extends TableDataProvider {
 
   override getItem(i: number): Item {
     const item = this.rows[i]
-    const data = dict(this.source.data)
 
     return item instanceof Group
       ? item as Item
-      : [...data.keys()].reduce((obj, name) => ({...obj, [name]: data.get(name)![item]}), {[DTINDEX_NAME]: item})
+      : Object.assign({[DTINDEX_NAME]: item}, this.source.get_row(item))
```

Comment:
Spread operator seems more idiomatic these days:

```suggestion
      : {[DTINDEX_NAME]: item, ...this.source.get_row(item)}
```